model_analyzer/monitor/nvml.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pynvml import *
+from multiprocessing.pool import ThreadPool
+import time
+
+from model_analyzer.monitor.record.memory import MemoryRecord
```

Comment:
The gathered data should be stored in a level that is accessible to all object, so it needs its own top level directory. In otherwords, a record is not unique to a monitor. I would move this up a level...

```
from model_analyzer.record.memory import MemoryRecord
from model_analyzer.record.collector import RecordCollector
```